src/vcpkg/build.cpp

```diff
@@ -849,6 +851,200 @@ namespace vcpkg::Build
+                state = State::None;
+                return; // don't capture output
+            }
+            else if (state == State::None)
+            {
+                print2(line, '\n');
+                return; // don't capture output
+            }
+            else if (state == State::Start && Strings::equals(line, Downloads::guid_marker_hash_mismatch_start))
+            {
+                state = State::Hashes;
+            }
+            else if (state == State::Hashes)
+            {
+                if (Strings::equals(line, Downloads::guid_marker_hash_mismatch_end))
+                    state = State::HaveHashes;
+                else if (Strings::starts_with(line, "expected="))
+                    current_entry.old_hash = line.substr(9 /* length of 'expected=' */);
+                else if (Strings::starts_with(line, "actual="))
+                    current_entry.new_hash = line.substr(7 /* length of 'actual=' */);
+            }
+            else if (state == State::HaveHashes &&
+                     line.find("Call Stack (most recent call first):") != std::string::npos)
+            {
+                state = State::CallStack;
+            }
+            else if (state == State::CallStack)
+            {
+                if (!line.empty())
+                {
+                    current_entry.complete_stack_trace += line;
+                    current_entry.complete_stack_trace += '\n';
+                    if (line.length() < 5) // a real entry can not only constis out of 5 chars
+                        return;
+                    parse_stack_trace_entry(line.substr(2));
+                }
+            }
+            captured_output += line;
+            captured_output += '\n';
+        }
+
+        void parse_stack_trace_entry(std::string line)
+        {
+            auto colon = line.find(":");
+            const auto printError = [&] {
+                print2(Color::error,
+                       "The Strack trace line should have the format `path/fileName.cmake:<line_number> "
+                       "(functionName)` but is ",
+                       line);
+            };
+            if (colon == std::string::npos)
+            {
+                printError();
+                return;
+            }
+            auto space = line.find(" ", colon);
+            if (space == std::string::npos)
+            {
+                printError();
+                return;
+            }
+            const auto path = line.substr(0, colon);
+            if (Strings::starts_with(path, "scripts/"))
+            {
+                return;
+            }
+            try
+            {
+                auto line_number = std::stoi(line.substr(colon + 1, space));
+                current_entry.stack_trace.push_back(Entry::StackTraceEntry{path, line_number});
+            }
+            catch (std::exception&)
+            {
+                printError();
+                return;
+            }
+        }
```

Comment:
why are we parsing the stack trace still?

Reply:
Because we have to know where the SHA512 value was given to a function so that we can replace the value.

Reply:
And I have not found another way of getting a stacktrace

Reply:
ah, I see; is there anything wrong with literally just find-replace'ing the sha512 itself in the entire file? That would also fix stuff like:

```cmake
set(hash ...)
vcpkg_download_distfile(... SHA512 "${hash}")
```

Reply:
We could do that for the non 0 case. But I have never seen that and I don't know why one should do that, but I can implement that :D

Reply:
Hm ok it is done ~25 times in the vcpkg repo

Reply:
Yeah, it's pretty common.

Reply:
"pretty common": 25 ports out of 1764, 1.4%

Reply:
fair point.

Reply:
The 
```cmake
set(hash ...)
vcpkg_download_distfile(... SHA512 "${hash}")
```
case now also works